Detect_Range_Readings.c

```diff
@@ -0,0 +1,95 @@
+#include <stdio.h>
+#include <assert.h>
+#include <stdlib.h>
+#include "Detect_Range_Readings.h"
+
+void sort_array(int *array, int arraysize)
+{
+    int i, j, min_idx; 
+  
+    // One by one move boundary of 
+    // unsorted subarray 
+    for (i = 0; i < arraysize-1; i++) 
+    { 
+        
+        // Find the minimum element in 
+        // unsorted array 
+        min_idx = i; 
+        for (j = i+1; j < arraysize; j++) 
+        {
+             if (array[j] < array[min_idx]) 
+             {
+                min_idx = j; 
+             }
+        }
+        // Swap the found minimum element 
+        // with the first element 
+        swap(&array[min_idx], &array[i]); 
+    } 
+}
+
+//Swap function
+void swap(int *xp, int *yp) 
+{ 
+    int temp = *xp; 
+    *xp = *yp; 
+    *yp = temp; 
+}
+
+//Function to print an array
+void printArray(int array[], int size) 
+{ 
+    int i; 
+    printf("The charging current value during the process of charging is\n");
+    for (i=0; i < size; i++) 
+    {
+        printf("%d ", array[i]);
+    }
+     
+    printf("\n");  
+    printf("\n"); 
+}
+
+int check_readings(int *array, int size)
+{
+    static int count=0;
+    int i,flag;
+    
+    for(i=0; i<size; i++)
+    {
+        if ((array[i] == array[i+1]))
```

Comment:
accessing i+1 element when i is pointing to last element could return a normal integer or a garbage value or a runtime exception.
The behavior is undefined.
The design should take care of not accessing unwanted memory, so that behavior of code is consistent.